src/main/java/hudson/remoting/Channel.java

```diff
@@ -829,8 +830,16 @@ public void terminate(IOException e) {
             } // JENKINS-14909: leave synch block
         } finally {
             if (e instanceof OrderlyShutdown) e = null;
-            for (Listener l : listeners.toArray(new Listener[0]))
-                l.onClosed(this, e);
+            for (Listener l : listeners.toArray(new Listener[0])) {
+                try {
+                    l.onClosed(this, e);
+                } catch (Throwable t) {
+                    LogRecord lr = new LogRecord(Level.SEVERE, "Listener {0} propagated an exception for channel {1}'s close: {2}");
+                    lr.setThrown(t);
+                    lr.setParameters(new Object[]{l, this, t.getMessage()});
```

Comment:
Yes because sometimes the exception can get separated from the message, also some people use custom formatters that drop the stack trace (bold people)